src/Serilog.Sinks.AzureDataExplorer.Tests/AzureDataExplorerSinkE2ETests.cs

```diff
@@ -0,0 +1,295 @@
+using Kusto.Data;
+using Kusto.Data.Net.Client;
+using Kusto.Data.Common;
+using Serilog.Sinks.AzureDataExplorer.Extensions;
+
+namespace Serilog.Sinks.AzureDataExplorer.Tests;
+
+/*
+ * This is an End to End Testcase which requires the following input to be set as environment variables
+ * ingestionURI : ingestion URL of ADX
+ * databaseName : database name 
+ * bufferFileName : If durable mode is required, we need to mention the buffer file name
+ * appId : Application client Id
+ * appKey : Application client key
+ * tenant : Authority
+ * These E2E testcases are disabled by default,
+ * to enable it
+ * 1. please change the access specifier of this class (AzureDataExplorerSinkE2ETests) to public
+ * 2. remove the System.Diagnostics.CodeAnalysis.SuppressMessage
+ */
+
+[System.Diagnostics.CodeAnalysis.SuppressMessage("Usage", "xUnit1000:Test classes must be public", Justification = "Disabled")]
+internal class AzureDataExplorerSinkE2ETests : IDisposable
+{
+    private readonly string m_generatedBufferFileName;
+    private readonly string m_generatedTableName;
+
```

Comment:
There is a lot of simarity of code between the 2 tests, would it be possible that we have different [Theory] in the tests since there is so much common code. Makes it easier to maintain